### PR TITLE
queue_specific_queue is missing its do_push vtable entry

### PR DESCRIPTION
**Pull request by MadCoder**
Also reorder all vtable entries to be in the same order (push, invoke,
wakeup).

Reenable dispatch_context_for_key.

SR: SR-4039

### CODE REVIEW

**Comment by MadCoder** [comment]
@swift-ci please test

File: src/init.c [hunk 1]
```diff
@@ -406,8 +407,8 @@ DISPATCH_VTABLE_SUBCLASS_INSTANCE(queue_main, queue,
 	.do_type = DISPATCH_QUEUE_SERIAL_TYPE,
 	.do_kind = "main-queue",
 	.do_dispose = _dispatch_queue_dispose,
-	.do_invoke = _dispatch_queue_invoke,
 	.do_push = _dispatch_queue_push,
+	.do_invoke = _dispatch_queue_invoke,
```
Comment (das) [inline src/init.c]: FWIW push -> wakeup -> invoke would have seemed the natural order to me ;-)

Comment (MadCoder) [inline src/init.c]: I just kept the order that was dominating. For real the better order is:

wakeup -> push -> invoke. (cause it's in that order it happens)

Comment (MadCoder) [inline src/init.c]: actually no I'm tired, it's push on self, wakeup onto target, then invoke you're correct :P

**Pull request merged**